test/digraph/cycles/karp.jl

```diff
@@ -0,0 +1,107 @@
+@testset "Karp Minimum Cycle Mean" begin
+     g = DiGraph(12)
+     w = [0. 0.  0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 0. 0.  1.1 0.0  0.  0.
+          0. 1.2 0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 1.  0. 0. 0. 0. 0. 0.  1.3 0.0  0.  0.
+          0. 1.2 0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 1.  0. 0. 0. 0. 0. 0.  1.3 0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 0. 0.  0.  0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 0. 0.  0.  0.0  0.4 0.
+          0. 0.  0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 1. 0.  0.  0.0  0.  0.
+          0. 0.  0. 0. 0. 0. 0. 0.  1.2 0.0  0.  0.]
+     add_edge!(g, 1, 7)
+     add_edge!(g, 2, 7)
+     add_edge!(g, 3, 9)
+     add_edge!(g, 4, 2)
+     add_edge!(g, 4, 7)
+     add_edge!(g, 5, 2)
+     add_edge!(g, 5, 9)
+     add_edge!(g, 6, 2)
+     add_edge!(g, 6, 7)
+     add_edge!(g, 7, 2)
+     add_edge!(g, 7, 9)
+     add_edge!(g, 9, 11)
+     add_edge!(g, 10, 7)
+     add_edge!(g, 11, 7)
+     add_edge!(g, 12, 9)
+
+     c, λ = karp_minimum_cycle_mean(g, w)
```

Comment:
make sure you cycle through all graph types: rename `g` to `g1` and do
```
for g in maketestgraphs(g1)
  c, λ = @inferred(karp_minimum_cycle_mean(g, w))
     @test c == [9, 11, 7]
     @test λ == 0.9
end
```

(repeat for the other tests as well)